widget/embedded/src/store/wallets.ts

```diff
@@ -54,156 +56,189 @@ interface WalletsStore {
 
 export const useWalletsStore = createSelectors(
   create<WalletsStore>()(
-    subscribeWithSelector((set, get) => ({
-      connectedWallets: [],
-      balances: {},
-      loading: false,
-      connectWallet: (accounts, findToken) => {
-        const getWalletsDetails = get().getWalletsDetails;
-        set((state) => ({
-          loading: true,
-          connectedWallets: state.connectedWallets
-            .filter((wallet) => wallet.walletType !== accounts[0].walletType)
-            .concat(
-              accounts.map((account) => {
-                const shouldMarkWalletAsSelected = !state.connectedWallets.some(
-                  (connectedWallet) =>
-                    connectedWallet.chain === account.chain &&
-                    connectedWallet.selected &&
-                    /**
-                     * Sometimes, the connect function can be called multiple times for a particular wallet type when using the auto-connect feature.
-                     * This check is there to make sure the chosen wallet doesn't end up unselected.
-                     */
-                    connectedWallet.walletType !== account.walletType
+    subscribeWithSelector((set, get) => {
+      return {
+        connectedWallets: [],
+        balances: {},
+        loading: false,
+        connectWallet: (accounts, findToken) => {
+          eventEmitter.emit(MainEvents.WalletEvent, {
+            type: WalletEventTypes.CONNECTED,
+            payload: { walletType: accounts[0].walletType, accounts },
+          });
```

Comment:
I think it's better to move the emitters from the store to the `wallets.tsx` file where the connect and disconnect functions are invoked.

Reply:
I disagree with that for some reasons: 1. In the future if we try to use connect and disconnect in another place in the project, we should consider handling events in that place too. 2. I think it's better to keep it consistent with quote store.